From 8506279fbc0bba9cb83110dd1ec954fdb235ece3 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 18 May 2022 13:04:17 +0300
Subject: [PATCH] [SW-0] gaudi3: remove fw mem iatu configuration

In recovery mode, secured FW upgrade will be done using SRAM
which will come up in mode 0. Currently FW memory is also configured
to map to bar 4/5. Since FW memory wont be used for secured FW upgrade,
this patch remove FW memory iatu configuration.

Change-Id: Ie8f5210badba67217408e93f2633cfa6d99dce23
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 8ffbded..928768d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -187,11 +187,6 @@
 #define BMU_MEM_0_SRC_END			0xFF
 #define BMU_MEM_0_DST_OFFSET			0xFFFFF00
 
-#define BMU_MEM_1_DST_BASE			0x300
-#define BMU_MEM_1_SRC_START			0x0
-#define BMU_MEM_1_SRC_END			0xFF
-#define BMU_MEM_1_DST_OFFSET			0x7FD00
-
 void pcie_config_iatu(void)
 {
 	uint32_t val, dbi_val;
@@ -231,18 +226,9 @@
 	/* bits[47:20] of 0x0200_FFFF_F000_0000 */
 	regmap_writel(BMU_MEM_0_DST_OFFSET, mmD0_PCIE_BMU_MEM_DST_OFFS_0);
 
-	/* Region 1 MEM region FW MEM BAR 4/5 */
-	/* upper 16 bits of 0x0300_007F_D000_0000 */
-	regmap_writel(BMU_MEM_1_DST_BASE, mmD0_PCIE_BMU_MEM_DST_BASE_1);
-	regmap_writel(BMU_MEM_1_SRC_START, mmD0_PCIE_BMU_MEM_SRC_START_1);
-	regmap_writel(BMU_MEM_1_SRC_END, mmD0_PCIE_BMU_MEM_SRC_END_1);
-	/* bits [47:20] of 0x0300_007F_D000_0000 */
-	regmap_writel(BMU_MEM_1_DST_OFFSET, mmD0_PCIE_BMU_MEM_DST_OFFS_1);
-
 	/* Enable Region 0 CFG and MEM and Region 1 MEM */
 	val = regmap_readl(mmD0_PCIE_BMU_CFG_EN);
-	val |= D0_PCIE_BMU_CFG_EN_F0_CFG_R0_M | D0_PCIE_BMU_CFG_EN_F0_MEM_R0_M |
-		D0_PCIE_BMU_CFG_EN_F0_MEM_R1_M;
+	val |= D0_PCIE_BMU_CFG_EN_F0_CFG_R0_M | D0_PCIE_BMU_CFG_EN_F0_MEM_R0_M;
 	regmap_writel(val, mmD0_PCIE_BMU_CFG_EN);
 
 	/* Enable BMU */
